### PR TITLE
Do not allow to update an instance if the instance is failed or is being deprovisioned

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- return http 422 if the instance is deproviosioned or failed

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/broker/instance_update.go [hunk 1]
```diff
@@ -105,6 +105,22 @@ func (b *UpdateEndpoint) Update(_ context.Context, instanceID string, details do
 		logger.Errorf("cannot fetch provisioning lastProvisioningOperation for instance with ID: %s : %s", instance.InstanceID, err.Error())
 		return domain.UpdateServiceSpec{}, errors.New("unable to process the update")
 	}
+	if lastProvisioningOperation.State == domain.Failed {
+		return domain.UpdateServiceSpec{}, apiresponses.NewFailureResponse(errors.New("Unable to process an update of a failed instance"), http.StatusUnprocessableEntity, "awer")
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/broker/instance_update.go:109]: nit: not sure if `awer` is intentional logging message, I wouldn't know what that means but seems to be unique in KEB :)

**Review by wozniakjan — APPROVED** [review]
/lgtm
/hold

feel free to unhold if `awer` was intentional :)

File: components/kyma-environment-broker/internal/broker/instance_update.go (hunk 1 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/broker/instance_update.go]: it put it by accident, fixed

**Review by wozniakjan — APPROVED** [review]
/lgtm
/unhold

**Pull request merged**